DirectProgramming/DPC++FPGA/Tutorials/Features/scheduler_target_fmax/src/scheduler_target_fmax.cpp

```diff
@@ -0,0 +1,164 @@
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities/include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+using namespace sycl;
+
+constexpr unsigned seed = 1313;
+
+// Forward declare the kernel names in the global scope.
+// This FPGA best practice reduces name mangling in the optimization reports.
+class Default;
+class Fmax480Attr;
+class Fmax240Attr;
+class Fmax240IIAttr;
+
+// Runs the Kernel
+void KernelRun(size_t size, const std::vector<char> &input_data,
+               std::vector<unsigned> &output_data) {
+#if defined(FPGA_EMULATOR)
+  ext::intel::fpga_emulator_selector device_selector;
+#else
+  ext::intel::fpga_selector device_selector;
+#endif
+
+  try {
+    // create the SYCL device queue
+    queue q(device_selector, dpc_common::exception_handler,
+            property::queue::enable_profiling{});
+
+    buffer input_buffer(input_data);
+    buffer output_buffer(output_data);
+
+    q.submit([&](handler &h) {
+      accessor input_a(input_buffer, h, read_only);
+      accessor output_a(output_buffer, h, write_only, no_init);
+
+      h.single_task<Default>([=]() [[intel::kernel_args_restrict]] {
+        unsigned hash = 0;
+        for (size_t i = 0; i < size; i++) {
+          hash = (hash * seed) + input_a[i];
+        }
+        output_a[0] = hash;
+      });
+    });
+
+    q.submit([&](handler &h) {
+      accessor input_a(input_buffer, h, read_only);
+      accessor output_a(output_buffer, h, write_only, no_init);
+
+      h.single_task<Fmax480Attr>([=]()
+                                     [[intel::kernel_args_restrict,
+                                       intel::scheduler_target_fmax_mhz(480)]] {
+                                       unsigned hash = 0;
+                                       for (size_t i = 0; i < size; i++) {
+                                         hash = (hash * seed) + input_a[i];
+                                       }
+                                       output_a[1] = hash;
+                                     });
+    });
+
+    q.submit([&](handler &h) {
+      accessor input_a(input_buffer, h, read_only);
+      accessor output_a(output_buffer, h, write_only, no_init);
+
+      h.single_task<Fmax240Attr>([=]()
+                                     [[intel::kernel_args_restrict,
+                                       intel::scheduler_target_fmax_mhz(240)]] {
+                                       unsigned hash = 0;
+                                       for (size_t i = 0; i < size; i++) {
+                                         hash = (hash * seed) + input_a[i];
+                                       }
+                                       output_a[2] = hash;
+                                     });
+    });
+
+    q.submit([&](handler &h) {
+      accessor input_a(input_buffer, h, read_only);
+      accessor output_a(output_buffer, h, write_only, no_init);
+
+      h.single_task<Fmax240IIAttr>([=
```

Comment:
Would help to work with clang-format to try and get the line-breaks better here.

Reply:
Addressed. Shortened kernel names to get better clang-format results.